status/discover.go

```diff
@@ -173,26 +222,37 @@ func DiscoverGrafana() string {
 		if err == nil {
 			parts := strings.Split(parsedURL.Hostname(), ".")
 			if len(parts) >= 2 {
-				routeURL := discoverServiceURL(parts[1], parts[0])
+				routeURL, err := discoverServiceURL(parts[1], parts[0])
+				if err != nil {
+					log.Debugf("[GRAFANA] URL discovery failed: %v", err)
+				}
 				appstate.GrafanaDiscoveredURL = strings.TrimSuffix(routeURL, "/")
 			}
 		}
 	}
 	return appstate.GrafanaDiscoveredURL
 }
 
-func discoverServiceURL(ns, service string) (url string) {
-	log.Debugf("URL discovery for service '%s', namespace '%s'...", service, ns)
+func discoverServiceURL(ns, service string) (url string, err error) {
+	log.Debugf("[%s] URL discovery for service '%s', namespace '%s'...", strings.ToUpper(service), service, ns)
+	url = ""
 	client, err := getClient()
 
+	// If the client is not openshift return and avoid discover
 	if err != nil {
-		log.Debugf("Discovery failed: %v", err)
+		log.Debugf("[%s] Discovery failed: %v", strings.ToUpper(service), err)
+		return
+	}
+
+	if !client.IsOpenShift() {
+		err = errors.New("Client is not Openshift, discovery url is only supported in Openshift")
```

Comment:
For more consistency, can we print that error directly here instead of returning it and relying on the caller? Except if something particular is done with that error beside logging, but I don't think it's the case.

I remember that I removed the "error" returns from this function (and have them printed directly from here instead), in order to have a consistent logging between grafana and jaeger.

Reply:
So the `error` is still returned and I don't think we need it (do we?) Which makes eventual errors still being logged twice, first in the function, then from the caller.

Reply:
The logging has been removed from the call for jaeger discovery, but it's still there for grafana, so still being logged twice